core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala

```diff
@@ -885,8 +885,10 @@ class AuthorizerIntegrationTest extends BaseRequestTest {
   def shouldThrowTransactionalIdAuthorizationExceptionWhenNoTransactionAccessOnEndTransaction(): Unit = {
     addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Write)), producerTransactionalIdResource)
     addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Write)), Resource.ProducerIdResource)
+    addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Write)), topicResource)
```

Comment:
Should we also add a test case where there is a real authorization failure and yet the transaction can be aborted?